### PR TITLE
Expose Ivy's circular dependency check via UpdateOptions

### PR DESCRIPTION
**Pull request by eed3si9n**
See http://ant.apache.org/ivy/history/2.3.0/concept.html#circular

By default circular dependencies are warned, but they do not halt the dependency resolution. Using the following setting, circular dependencies can be treated as an error.

``` scala
updateOptions := updateOptions.value.withCircularDependencyLevel(CircularDependencyLevel.Error)
```

### CODE REVIEW

File: ivy/src/main/scala/sbt/CircularDependencyLevel.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package sbt
+
+import org.apache.ivy.plugins.circular.{ CircularDependencyStrategy, WarnCircularDependencyStrategy, IgnoreCircularDependencyStrategy, ErrorCircularDependencyStrategy }
+
+/**
+ * Wrapper around circular dependency strategy.
+ */
+trait CircularDependencyLevel {
+  private[sbt] def ivyStrategy: CircularDependencyStrategy
```
Comment (jsuereth) [inline ivy/src/main/scala/sbt/CircularDependencyLevel.scala:9]: Do you know the ramifications of a publicly extendable class where bits are marked `private[sbt]`?  Would extenders of this trait need to be in the sbt package?

**Comment by jsuereth** [comment]
LGTM

File: ivy/src/main/scala/sbt/CircularDependencyLevel.scala (hunk 1 above)
Comment (eed3si9n) [inline ivy/src/main/scala/sbt/CircularDependencyLevel.scala:9]: I guess I should seal that sucker.

**Comment by eed3si9n** [comment]
sealed the `CircularDependencyLevel` trait.

**Pull request merged**